src/granite/table.cr

```diff
@@ -9,11 +9,37 @@ macro disable_granite_docs?(stmt)
 end
 
 module Granite::Table
-  macro included
-    macro inherited
-      disable_granite_docs? SETTINGS = {} of Nil => Nil
-      disable_granite_docs? PRIMARY = {name: id, type: Int64, auto: true}
+  module ClassMethods
+    def primary_name : String?
+      {% begin %}
+      {% primary_key = @type.instance_vars.find { |ivar| (ann = ivar.annotation(Granite::Column)) && ann[:primary] } %}
+      {% if pk = primary_key %}
+        {{pk.name.stringify}}
+      {% end %}
+    {% end %}
     end
+
+    def quoted_table_name : String
+      adapter.quote(table_name)
+    end
+
+    def quote(column_name) : String
+      adapter.quote(column_name)
+    end
+
+    # Returns the name of the table for `self`
+    # defaults to the model's name underscored + 's'.
+    def table_name : String
+      {% begin %}
+        {% table_ann = @type.annotation(Granite::Model) %}
+        {{table_ann && !table_ann[:table].nil? ? table_ann[:table] : @type.name.underscore.stringify}}
+      {% end %}
+    end
+  end
+
+  macro table_name(name)
+    @[Granite::Model(table: {{(name.is_a?(StringLiteral) ? name : name.stringify) || nil}})]
```

Comment:
hmm.  Seems like the macro and annotation should match here as well but not sure what name works for both.

Reply:
Model makes it flexible enough if we wanted to add other fields to it in the future.

Reply:
But thats a good question of what would happen if we had another macro that did something else.  

What about making the macro like `model table: "people"`, then it could support extra stuff if anything else makes sense.

Reply:
I like `table` and `column` as the macros and annotation names.  Maybe consider `database` instead of adapter.

Reply:
I could live with that, we'll just have to add more annotations if we ever think of something else we would use.